### ISSUE TITLE
Loader Cleanups

### ISSUE THREAD

**Issue by sjmiles** [PolymerLabs/arcs#3757]
Loader infrastructure needs cleanup.

Features that are now expected of loader (e.g. url remapping) were originally only used by Shells, and are not well integrated into runtime, complicating the class structure. I believe this can be fixed with basic refactoring.

Also the following features do not fit neatly into the current Loader structure, and feature implementers have had to use workarounds (also likely fixable with basic refactoring).
  - dev-shell needs to directly return buffers for certain paths (from memory)
  - WASM loader needs to return ArrayBuffer

Currently the loadWasmBinary is implemented to use `fetch` for paths beginning with `https://`, and to otherwise use `fs`. Therefore apps fail under WebView when trying to load simple paths (because `fs` doesn't exist under WebView). Suggested work-around (until this ticket is resolved) is to use paths prefixed with `file://` and modify loadWasmBinary to detect that protocol and delegate to `fetch`.

**Comment by piotrswigon** [comment]
I have verified that the WebView in the Demo App can load wasm particles.

Here is the change I did and which loaded the wasm particle: https://github.com/piotrswigon/arcs/commit/00895805c713bbd53701816d91147badece74365

We currently don't have a convenient way of addressing wasm particles in recipes and switching between loading from the APK and from the workstation, but these are different issues.

@sjmiles feel free to make the loader improvements or close this issue if you think there's nothing here.
